packages/flutter_tools/lib/src/test/flutter_web_platform.dart

```diff
@@ -56,10 +56,16 @@ shelf.Handler createDirectoryHandler(Directory directory) {
       return shelf.Response.notFound('Not Found');
     }
     final String? contentType = resolver.lookup(file.path);
+    final bool needsCrossOriginIsolated = crossOriginIsolated && uriPath.endsWith('.html');
     return shelf.Response.ok(
       file.openRead(),
       headers: <String, String>{
-        if (contentType != null) 'Content-Type': contentType
+        if (contentType != null) 'Content-Type': contentType,
+        if (needsCrossOriginIsolated)
+          ...<String, String>{
+            'Cross-Origin-Opener-Policy': 'same-origin',
+            'Cross-Origin-Embedder-Policy': 'require-corp',
```

Comment:
What if we always used these, in all modes?

Reply:
I considered this. However, that would be a breaking change for anyone who writes a test that relies on being in a non-`crossOriginIsolated` context. It seems like it would be weird to do a cross-origin fetch in a unit test, but I could see it in an integration test possibly. I tried to make this change as non-breaking as possible. If we want to re-evaluate that decision later and say "all `flutter test` runs should run in a `crossOriginIsolated` context and if that breaks your test you should change your test" I think I'd be okay with that, but I'd rather not overload that behavior change with this PR.